components/CreateOrUpdateRecipeForm.js

```diff
@@ -0,0 +1,232 @@
+import styled from "styled-components";
+import Link from "next/link";
+
+export default function CreateOrUpdateRecipeForm({
+  errorMessage,
+  inputValidation,
+  recipeDetails = {},
+  onSubmit,
+}) {
+  const formTitle =
+    Object.keys(recipeDetails).length === 0
+      ? "Create new recipe"
+      : "Update Recipe";
+  const formButtonTitle =
+    Object.keys(recipeDetails).length === 0 ? "Add Recipe" : "Save Changes";
+  const abortLinkUrl =
+    Object.keys(recipeDetails).length === 0
+      ? "/"
+      : `/recipe/${recipeDetails.id}`;
+
+  return (
+    <>
+      <h1>{formTitle}</h1>
+
+      {errorMessage ? (
+        <StyledErrorMessageContainer>
+          <StyledErrorMessage>{errorMessage}</StyledErrorMessage>
+        </StyledErrorMessageContainer>
+      ) : (
+        ""
+      )}
```

Comment:
If we only want to show an element based on the value of a variable, otherwise show nothing, then this might be a simpler solution. 

```suggestion
      {errorMessage && (
        <StyledErrorMessageContainer>
          <StyledErrorMessage>{errorMessage}</StyledErrorMessage>
        </StyledErrorMessageContainer>
      )}
```